src/factory/AnotherCloneFactory.sol

```diff
@@ -333,12 +375,17 @@ contract AnotherCloneFactory is AccessControlUpgradeable {
      * @notice
      *  Predict the new ERC721AB collection address
      *
+     * @param _implementationId implementation identifier
      * @param _salt address of the new implementation contract
      *
      * @return _predicted predicted address for the given `_salt`
      */
-    function predictERC721Address(bytes32 _salt) external view returns (address _predicted) {
-        _predicted = Clones.predictDeterministicAddress(erc721Impl, _salt, address(this));
+    function predictERC721Address(uint256 _implementationId, bytes32 _salt)
+        external
+        view
+        returns (address _predicted)
+    {
+        _predicted = Clones.predictDeterministicAddress(erc721ImplAddresses[_implementationId], _salt, address(this));
```

Comment:
should we revert here if `erc721ImplAddresses[_implementationId]` is undefined or let `predictDeterministicAddress` do it?

Reply:
**if we pass an undefined `_implementationId`, reading the array `erc721ImplAddresses` will revert (panic out of bond index). 

so in order to save gas, i did not add that check ( _implementationId >= erc721ImplAddresses.length)

the downside is that the revert message is a bit generic and not customized but i feel like its worth the few units of gas saved.**